mmseg/core/utils/layer_decay_optimizer_constructor.py

```diff
@@ -0,0 +1,165 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+import json
+
+from mmcv.runner import (OPTIMIZER_BUILDERS, DefaultOptimizerConstructor,
+                         get_dist_info)
+
+from ...utils import get_root_logger
+
+
+def get_num_layer_layer_wise(var_name, num_max_layer=12):
+    """Get the layer id to set the different learning rates in ``layer_wise``
+    decay_type.
+
+    Args:
+        var_name (str): The key of the model.
+        num_max_layer (int): Maximum number of backbone layers.
+
+    Returns:
+        int: The id number corresponding to different
+        learning rate in ``LearningRateDecayOptimizerConstructor``.
+    """
+
+    if var_name in ('backbone.cls_token', 'backbone.mask_token',
+                    'backbone.pos_embed'):
+        return 0
+    elif var_name.startswith('backbone.downsample_layers'):
+        stage_id = int(var_name.split('.')[2])
+        if stage_id == 0:
+            layer_id = 0
+        elif stage_id == 1:
+            layer_id = 2
+        elif stage_id == 2:
+            layer_id = 3
+        elif stage_id == 3:
+            layer_id = num_max_layer
+        return layer_id
+    elif var_name.startswith('backbone.stem_norm'):
+        layer_id = 0
+        return layer_id
+    elif var_name.startswith('backbone.downsample_norm'):
+        stage_id = int(
+            var_name.split('backbone.downsample_norm')[1].split('.')[0])
+        if stage_id == 0:
+            layer_id = 2
+        elif stage_id == 1:
+            layer_id = 3
+        elif stage_id == 2:
+            layer_id = num_max_layer
+        return layer_id
```

Comment:
`backbone.stem_norm` and `backbone.downsample_norm` may not needed since they are not existed in mmcls convnext backbone. Same as in `get_num_layer_stage_wise()`.

Reply:
why did add 'if conditions' for `backbone.stem_norm` and `backbone.downsample_norm`,
from convnext official repo, there is not anything about these var_names.
https://github.com/facebookresearch/ConvNeXt/blob/3d444184dc27156e0562133c3b69b56f7efba500/semantic_segmentation/mmcv_custom/layer_decay_optimizer_constructor.py#L14-L42